### PR TITLE
update D-Scanner

### PR DESCRIPTION
**Pull request by ghost**
ping @thewilsonator for prioritized merge. Buildkite style check uses an outdated version and this must comes from here since the hash for phobos was already updated.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/expression.d [hunk 1]
```diff
@@ -2296,7 +2296,7 @@ extern (C++) final class StringExp : Expression
      * Returns:
      *      number of code units
      */
-    final size_t numberOfCodeUnits(int tynto = 0) const
+    size_t numberOfCodeUnits(int tynto = 0) const
```
Comment (wilzbach) [inline src/dmd/expression.d]: Why is this required?

Comment (BBasile) [reply src/dmd/expression.d]: the class is final. So are its member funcs.

Comment (BBasile) [reply src/dmd/expression.d]: to be perfectly clear: the useless_final_attribute check didn't pass.

**Review by RazvanN7 — APPROVED** [review]
